### PR TITLE
[PID ML] Separate models for different momentum

### PR DESCRIPTION
**Pull request by Lukaszz99**

### CODE REVIEW

File: Tools/PIDML/qaPidML.cxx [hunk 1]
```diff
@@ -321,9 +321,15 @@ struct pidml {
...
+    float pidLogits[3];
+    if (track.p() <= 0.5) {
+      pidLogits[3] = {model211TPC.applyModel(track), model2212TPC.applyModel(track), model321TPC.applyModel(track)};
+    } else {
+      pidLogits[3] = {model211All.applyModel(track), model2212All.applyModel(track), model321All.applyModel(track)};
+    }
     int pid = getParticlePdg(pidLogits);
-    if (pid == particlesPdgCode[i]) {
+    // condition for sign: we want to work only with pi, p and K, without antiparticles
+    if ((pid == particlesPdgCode[i]) & (track.sign() == 1)) {
```
Comment (saganatt) [inline Tools/PIDML/qaPidML.cxx:332]: use && and no brackets, that's more natural in ifs

File: Tools/PIDML/qaPidML.cxx [hunk 2]
```diff
@@ -332,19 +338,28 @@ struct pidml {
...
   }
 
-  PidONNXModel model211;
-  PidONNXModel model2212;
-  PidONNXModel model321;
+  // one model for one particle; Model with all TPC and TOF signal
+  PidONNXModel model211All;
+  PidONNXModel model2212All;
+  PidONNXModel model321All;
+  // Model with only TPC model
```
Comment (saganatt) [inline Tools/PIDML/qaPidML.cxx:345]: TPC signal

File: Tools/PIDML/qaPidML.cxx [hunk 3]
```diff
@@ -332,19 +338,28 @@ struct pidml {
...
+  PidONNXModel model211TPC;
+  PidONNXModel model2212TPC;
+  PidONNXModel model321TPC;
 
-  Configurable<bool> cfgUseTOF{"useTOF", true, "Use ML model with TOF signal"};
+  // Configurable<bool> cfgUseTOF{"useTOF", true, "Use ML model with TOF signal"};
```
Comment (saganatt) [inline Tools/PIDML/qaPidML.cxx:350]: Just delete this line

File: Tools/PIDML/qaPidML.cxx [hunk 4]
```diff
@@ -321,9 +321,15 @@ struct pidml {
   template <std::size_t i, typename T>
   void pidML(const T& track, const int pdgCodeMC)
   {
-    float pidLogits[3] = {model211.applyModel(track), model2212.applyModel(track), model321.applyModel(track)};
+    float pidLogits[3];
+    if (track.p() <= 0.5) {
```
Comment (saganatt) [inline Tools/PIDML/qaPidML.cxx:325]: is this in agreement with the training setup?

**Review by saganatt — CHANGES_REQUESTED** [review]

File: Tools/PIDML/qaPidML.cxx (hunk 4 above)
Comment (Lukaszz99) [reply Tools/PIDML/qaPidML.cxx]: I have added an array with momentum threshold values

**Review by saganatt — DISMISSED** [review]

**Review by saganatt — APPROVED** [review]

**Comment by saganatt** [comment]
macOS error unrelated

**Pull request merged**